app/src/main/java/com/example/sharewhatyoucanproject/rdashboard/rposts/PostListViewModel.kt

```diff
@@ -0,0 +1,64 @@
+package com.example.sharewhatyoucanproject.rposts
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.example.sharewhatyoucanproject.models.GeoPoint
+import com.example.sharewhatyoucanproject.models.PostModel
+import com.google.android.gms.tasks.OnCompleteListener
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlin.collections.ArrayList
+
+class PostListViewModel(
+    private val db: FirebaseFirestore,
+) : ViewModel() {
+    private val arrayList: ArrayList<PostModel> = mutableListOf<PostModel>() as ArrayList<PostModel>
+    private val _taskResult = MutableLiveData<TaskResult>()
+    val taskResult: LiveData<TaskResult> = _taskResult
+
+    fun getlist() {
+        db.collection("posts")
+            .whereEqualTo("status", 0)
+            .get()
+            .addOnCompleteListener(
+                OnCompleteListener { task ->
+                    if (task.isSuccessful) {
+                        for (postResult in task.result) {
+                            val postModel = postResult.getGeoPoint("location")?.let {
+                                PostModel(
+                                    postResult.get("imageUrl").toString(),
+                                    "" + postResult.getString("title").toString(),
```

Comment:
Since you are calling toString you don't need to manually add an empty string, so you can remove this: `"" +`

Reply:
-done